### PR TITLE
Feat/staking vault init

### PR DESCRIPTION
**Pull request by loga4**
- add a new VAULT_REGISTRY_ROLE role
- move storage to erc7201 format
- add accounting initializer
- refactor StakingVault initialization

### CODE REVIEW

File: contracts/0.8.25/Accounting.sol [hunk 1]
```diff
@@ -88,15 +88,23 @@ contract Accounting is VaultHub {
...
-    ) VaultHub(_admin, _lido, _treasury) {
+    ) VaultHub(_lido, _treasury) {
         LIDO_LOCATOR = _lidoLocator;
         LIDO = _lido;
     }
 
+    function initialize(address _admin) external initializer {
+        if (_admin == address(0)) revert ZeroArgument("_admin");
+
+        __AccessControlEnumerable_init();
```
Comment (folkyatina) [inline contracts/0.8.25/Accounting.sol]: Do we really need this?

Anyway, maybe move it to __VaultHub_init as Accounting does nothing with access control?
As well as the initial role grant?

File: contracts/0.8.25/vaults/StakingVault.sol [hunk 2]
```diff
@@ -39,32 +38,34 @@ contract StakingVault is IStakingVault, IBeaconProxy, VaultBeaconChainDepositor,
...
+    modifier onlyBeacon() {
+        if (msg.sender != getBeacon()) revert UnauthorizedSender(msg.sender);
+        _;
     }
 
     /// @notice Initialize the contract storage explicitly.
     ///         The initialize function selector is not changed. For upgrades use `_params` variable
     ///
-    /// @param _owner owner address that can TBD
+    /// @param _owner vaultStaffRoom address
```
Comment (folkyatina) [inline contracts/0.8.25/vaults/StakingVault.sol]:
```suggestion
    /// @param _owner vault owner address
```
basic vault knows nothing about upper layers

File: contracts/0.8.25/vaults/StakingVault.sol [hunk 3]
```diff
@@ -228,5 +229,5 @@ contract StakingVault is IStakingVault, IBeaconProxy, VaultBeaconChainDepositor,
     error NotHealthy();
     error NotAuthorized(string operation, address sender);
     error LockedCannotBeDecreased(uint256 locked);
-    error NonProxyCallsForbidden();
+    error UnauthorizedSender(address sender);
```
Comment (folkyatina) [inline contracts/0.8.25/vaults/StakingVault.sol]:
```suggestion
    error SenderShouldBeBeacon(address sender, address beacon);
```

File: contracts/0.8.25/vaults/VaultHub.sol [hunk 4]
```diff
@@ -46,52 +50,68 @@ abstract contract VaultHub is AccessControlEnumerableUpgradeable {
         uint16 treasuryFeeBP;
     }
 
-    /// @notice vault sockets with vaults connected to the hub
-    /// @dev first socket is always zero. stone in the elevator
-    VaultSocket[] private sockets;
-    /// @notice mapping from vault address to its socket
-    /// @dev if vault is not connected to the hub, its index is zero
-    mapping(IHubVault => uint256) private vaultIndex;
+    // keccak256(abi.encode(uint256(keccak256("VaultHub")) - 1)) & ~bytes32(uint256(0xff))
```
Comment (folkyatina) [inline contracts/0.8.25/vaults/VaultHub.sol:53]: do we have some linter/slither to check it?

**Review by folkyatina — APPROVED** [review]
Initialization order could be straightened a bit, imho

File: contracts/0.8.25/Accounting.sol (hunk 1 above)
Comment (loga4) [reply contracts/0.8.25/Accounting.sol]: agree

**Pull request merged**